hedera-node/hedera-admin-service/src/main/java/com/hedera/node/app/service/admin/FreezePreTransactionHandler.java

```diff
@@ -34,5 +35,5 @@ public interface FreezePreTransactionHandler extends PreTransactionHandler {
      *     com.hederahashgraph.api.proto.java.FreezeTransactionBody}
      * @return the metadata for the freeze
      */
-    TransactionMetadata preHandleFreeze(TransactionBody txn);
+    TransactionMetadata preHandleFreeze(TransactionBody txn, AccountID payer);
```

Comment:
I wonder, if we should pass an instance of `TransactionMetadata` to the prehandle-functions instead. `TransactionBody` and payer would already be set and the prehandle-methods would simple add additional information to the existing `TransactionMetadata`. In a few special cases, it has to create a new instance, but this will be the exception, I think.

Reply:
You mean it can be done in `PreHandleWorkflow` ?

Reply:
@tinker-michaelj @rbair23  Any thoughts ?

Reply:
> You mean we can create a SigTransactionMetadataBuilder with txn and payer set in PreHandleWorkflow and inject here right ? For Schedules we need to create ScheduleSigTransactionMetadataBuilder.
I think there will be more sub classes for metadata.

Sounds very reasonable to me @Neeharika-Sompalli

Reply:
Once we have the `createStore` method available, I will modify this to use `SigTransactionMetadataBuilder` as the constructor needs `KeyLookup` from PreHandleWorkflow.